web-server-lib/web-server/http/response.rkt

```diff
@@ -103,7 +106,10 @@
      #'(let ([out-id out-e])
          write-e ...)]))
 
-(define (output-response-head conn bresp [chunked? #f])
+(define (output-response-head conn raw-bresp [chunked? #f])
+  (define bresp ((current-header-handler)
+                 (struct-copy response raw-bresp
+                              [output void])))
```

Comment:
I think voiding the `output` here makes it so no response can output any data (see `output-response-body` below). It should be safe to pass the unmodified response to the handler proc, and it should be the handler proc's responsibility not to do anything silly with it.

Reply:
Oh, ignore this. I misread the surrounding code, not realizing this was only outputting the headers. Still, it might make sense to allow processors to alter the output, too.

Reply:
This `bresp` is used only inside this function. `response-output` in `output-response-body` will be the original one. It is unaffected.

Reply:
I replied too late :)

The reason why I only target `header` as opposed to the whole `response` is that `output-file` (which in turn calls `output-file/boundary`) creates a fake response with correct header but empty body, and then uses `output-response-head` to print the header directly. Giving an appearance that the `output` field could be modified would be misleading for such cases.